### DISCUSSION TITLE
Is there a way to cancel the loading process?

### DISCUSSION THREAD

**Discussion by Godelaune** [google/model-viewer#2591] [Q&A]
Hi,
I'd like to know if there is a way to cancel the loading-progress?
I figured out that even destroying all the DOM of model-viewer didn't stop it from loading the 3D-file in background.

Thank you for the good work and best regards!

**Comment by elalish** [comment]
Hmm, not that I'm aware of. How would you use a feature like this? You're the first to request it.

**Reply by Godelaune** [reply]
Hey, thanks for the quick answer. We use model-viewer in a webshop to display 3D-models of furniture. The 3D-models are quite large sometimes (40MB) and we'd like to give the customer a chance to cancel the loading progress if he is on a mobile data plan.

I found that cancelling the three.js loader is possible within the onprogress callback but I'm not sure if this has anything to do with the modelviewer loading progress.

For three.js:
```javascript
var loader = new THREE.STLLoader();

loader.load('test.stl', function ( geometry ) {
    //stuff here
}, function(url) {
    //logic here
    url.currentTarget.abort()
}) ;   
```
(from [stackoverflow](https://stackoverflow.com/questions/30584362/canceling-loading-an-object-into-a-scene-in-three-js?rq=1))

**Reply by elalish** [reply]
Hmm, that's a good point. I suppose really this means we should also show some UI to cancel. For back-compat we'll have to make it opt-in. Let's consider this a feature request.

**Reply by furkandindar** [reply]
hi, following up on this. I have a carousel of 3d items where user can change the item to display just like in the [webXR demo](https://modelviewer.dev/examples/augmentedreality/#webXR). I noticed that if you choose a different item before the selected item loads, it still loads the previous item and newly selected item. This also causes progress event to combine 2 model load together so when you change the model event.detail.totalProgress doesn't go back to 0.

**Reply by elalish** [reply]
Interesting - want to try a PR?
